packages/core/src/browser/frontend-connection-status.ts

```diff
@@ -0,0 +1,221 @@
+export class ConnectionStatusOptions {
+
+    static DEFAULT: ConnectionStatusOptions = {
+        requestTimeout: 1000,
+        retry: 5,
+        retryInterval: 2000,
+    };
+
+    /**
+     * Timeout for the HTTP GET request in milliseconds.
+     */
+    readonly requestTimeout: number;
+
+    /**
+     * Number of accepted timeouts.
+     */
+    readonly retry: number;
+
+    /**
+     * Retry interval in milliseconds.
+     */
+    readonly retryInterval: number;
+
+}
+
+@injectable()
+export class FrontendConnectionStatusService implements FrontendApplicationContribution {
+
+    private connectionState: ConnectionStateMachine;
+    // tslint:disable-next-line:no-any
+    private timer: any | undefined;
+
+    constructor(
+        @inject(ConnectionStatusOptions) @optional() protected readonly options: ConnectionStatusOptions = ConnectionStatusOptions.DEFAULT,
+        @inject(ContributionProvider) @named(ConnectionStatusContribution) protected readonly contributionProvider: ContributionProvider<ConnectionStatusContribution>
+    ) {
+        this.connectionState = new ConnectionStateMachine({ threshold: this.options.retry });
+    }
+
+    onStart() {
+        this.timer = setInterval(() => {
+            const xhr = new XMLHttpRequest();
+            xhr.timeout = this.options.requestTimeout;
+            xhr.onreadystatechange = event => {
+                const { readyState, status } = xhr;
+                if (readyState === XMLHttpRequest.DONE) {
+                    const success = status === 200;
+                    this.connectionState = this.connectionState.next(success);
+                    this.contributionProvider.getContributions().forEach(contribution => contribution.onStatusChange(this.connectionState));
+                }
+            };
+            xhr.onerror = () => {
+                this.connectionState = this.connectionState.next(false);
+                this.contributionProvider.getContributions().forEach(contribution => contribution.onStatusChange(this.connectionState));
+            };
+            xhr.open('GET', `${window.location.href}alive`);
+            try { xhr.send(); } catch { /* NOOP */ }
+        }, this.options.retryInterval);
+        this.contributionProvider.getContributions().forEach(contribution => contribution.onStatusChange(this.connectionState));
+    }
+
+    onStop() {
+        if (this.timer !== undefined) {
+            clearInterval(this.timer);
+            this.timer = undefined;
+        }
+    }
+
+}
+
+@injectable()
+export class ConnectionStatusStatusBarContribution implements ConnectionStatusContribution {
+
+    constructor( @inject(StatusBar) protected statusBar: StatusBar) {
+    }
+
+    onStatusChange(event: ConnectionStatusChangeEvent) {
+        this.statusBar.removeElement('connection-status');
+        const text = `$(${this.getStatusIcon(event.health)})`;
+        const tooltip = event.health ? `Connection health: ${event.health}%` : 'Not connected';
```

Comment:
`Online` and `Offline`? I think it is more understandable.

Reply:
I think you are absolutely right. I have changed it.